### PR TITLE
Added support for django's builtin bulk_update method if present

### PR DESCRIPTION
**Pull request by deeaarbee**
Fix for using django's builtin bulk_update method if available
[Link](https://github.com/Tesorio/django-anon/issues/48) for the issue

## Todos

- [x] Tests
- [x] Documentation
- [x] Changelog

### CODE REVIEW

File: anon/compat.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# deps
+from django_bulk_update.helper import bulk_update as ext_bulk_update
+
+
+def bulk_update(objects, manager, **bulk_update_kwargs):
+    """ Updates the list of objects using django queryset's
+        inbuilt `.bulk_update()` method if present, else
+        django_bulk_update's `bulk_update` will be used
```
Comment (caioariede) [inline anon/compat.py:8]: 📖 Could you please add parameters to the docstring?

[https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)

For example:

```python
"""
This is what this method does...

:param arg1: first argument
:type arg1: int
:param arg2: second argument
:type arg2: str
:return: A list of items
:rtype: list
"""
```

File: anon/compat.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# deps
+from django_bulk_update.helper import bulk_update as ext_bulk_update
+
+
+def bulk_update(objects, manager, **bulk_update_kwargs):
+    """ Updates the list of objects using django queryset's
+        inbuilt `.bulk_update()` method if present, else
```
Comment (caioariede) [inline anon/compat.py:7]: 📖 This is an invalid reStructuredText markup. Inline literals use double backticks:

[https://docutils.sourceforge.io/docs/user/rst/quickref.html](https://docutils.sourceforge.io/docs/user/rst/quickref.html)

File: anon/compat.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        django_bulk_update's `bulk_update` will be used
+    """
+    if getattr(manager, "bulk_update", None):
+        manager.bulk_update(objects, **bulk_update_kwargs)
+    else:
+        ext_bulk_update(objects, bulk_update_kwargs)
```
Comment (caioariede) [inline anon/compat.py:13]: 🎨 It's easier to ask for forgiveness than permission (EAFP):

[https://docs.python.org/3.4/glossary.html](https://docs.python.org/3.4/glossary.html)

```suggestion
    try:
        manager.bulk_update(objects, **bulk_update_kwargs)
    except AttributeError:
        ext_bulk_update(objects, bulk_update_kwargs)
```

File: anon/compat.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# deps
+from django_bulk_update.helper import bulk_update as ext_bulk_update
+
+
+def bulk_update(objects, manager, **bulk_update_kwargs):
```
Comment (caioariede) [inline anon/compat.py:5]: I think we need to add a unit test to this function

**Review by caioariede — CHANGES_REQUESTED** [review]
Thank you @deeaarbee it looks really good! Just some minor suggestions :)

File: anon/compat.py (hunk 1 above)
Comment (deeaarbee) [reply anon/compat.py]: I have added the parameters to the docstring. Please verify the format.

File: anon/compat.py (hunk 2 above)
Comment (deeaarbee) [reply anon/compat.py]: Thanks for the reference. I have updated the docstring.

File: anon/compat.py (hunk 4 above)
Comment (deeaarbee) [reply anon/compat.py:5]: Added unit test for the method. Please confirm if the test case makes sense.

File: anon/compat.py (hunk 3 above)
Comment (deeaarbee) [reply anon/compat.py]: Updated the code with your suggestion.

**Comment by caioariede** [comment]
That's awesome, thanks @deeaarbee :) The coverage is failing for other reasons as far as I can tell.

Can you please just add a new bullet under the **master** section in the `CHANGELOG`?

https://github.com/Tesorio/django-anon/blob/master/CHANGELOG.rst#master

Thanks again!

**Comment by caioariede** [comment]
@deeaarbee thank you for your contribution 🙏

**Pull request merged**